runtime/src/asset.rs

```diff
@@ -157,6 +179,70 @@ decl_module! {
         /// initialize the default event for this module
         fn deposit_event() = default;
 
+        /// This function is used to either register a new ticker or extend validity of an exisitng ticker
+        /// NB Ticker validity does not get carryforward when renewing ticker
+        ///
+        /// # Arguments
+        /// * `origin` It consist the signing key of the caller (i.e who signed the transaction to execute this function)
+        /// * `to_did` DID of the (future) owner of the ticker
+        /// * `_ticker` ticker to register
+        pub fn register_ticker(origin, to_did: IdentityId, _ticker: Vec<u8>) -> Result {
+            let _ = ensure_signed(origin)?;
+
+            let ticker = utils::bytes_to_upper(_ticker.as_slice());
+
+            ensure!(!<Tokens<T>>::exists(&ticker), "token already created");
+
+            let ticker_config = Self::ticker_registration_config();
+
+            ensure!(ticker.len() <= usize::try_from(ticker_config.max_ticker_length).unwrap_or_default(), "ticker length over the limit");
+
+            if Self::is_ticker_available(&ticker) {
+                // ticker not registered by anyone (or registry expired). we can charge fee and register this ticker
+            } else {
+                // ticker already registered to someone. Ensure that the ticker is registered to same did
+                ensure!(Self::is_ticker_registry_valid(&ticker, to_did), "ticker registered to someone else");
+            }
+            // charge fee
+            Self::charge_ticker_registration_fee(&ticker, to_did);
+
+            let now = <timestamp::Module<T>>::get();
+            let ticker_registration = TickerRegistration {
+                owner: to_did,
+                expiry: if let Some(exp) = ticker_config.registration_length { Some(now + exp) } else { None }
+            };
+
+            // Store ticker registration details
+            <Tickers<T>>::insert(&ticker, ticker_registration);
+
+            Ok(())
+        }
+
+        /// This function is used to transfer a ticker to someone else
+        ///
+        /// # Arguments
+        /// * `origin` It consist the signing key of the caller (i.e who signed the transaction to execute this function)
+        /// * `from_did` DID of the current owner of the ticker
+        /// * `to_did` DID of the future owner of the ticker
+        /// * `_ticker` ticker to transfer
+        pub fn transfer_ticker(origin, from_did: IdentityId, to_did: IdentityId, _ticker: Vec<u8>) -> Result {
```

Comment:
This needs to be a two-phase process - i.e. `to_did` needs to confirm they are willing to receive the ticker reservation. This seems a bit pedantic, but stops me reserving a ticker of "SILLY_NAME" and then transferring it to your identity. Happy to split this into a new story / task if you'd prefer.

Reply:
I was thinking about this when writing this function. I thought that it's not harmful to have a ticker registered to your name since a ticker does not have an inherent value. This is also why `register_ticker` does NOT verify that the sender is part of the owner did. Anyone can register a ticker for anyone (I was thinking about enabling white labelers at the time).

If we want this to be a 2 step process then I am happy to do it in this PR or as a new task. Whatever works best on the Product side. I am fine with either approach. Just let me know how you want to proceed :)

Reply:
Agreed - I think the reservation process should also check that the origin is a signing key of the DID receiving the ticker. And when transferring tickers, receiving DID should need to accept the transfer.

Reply:
Data storage should be such that you can easily determine the list of "pending authorisation" requests for tickers.